pkg/controller/clusterdeployment/clusterprovisions.go

```diff
@@ -175,10 +175,11 @@ func (r *ReconcileClusterDeployment) startNewProvision(
 		},
 	}
 
-	// Copy over the cluster ID and infra ID from previous provision so that a failed install can be removed.
-	if cd.Spec.ClusterMetadata != nil {
-		provision.Spec.PrevClusterID = &cd.Spec.ClusterMetadata.ClusterID
-		provision.Spec.PrevInfraID = &cd.Spec.ClusterMetadata.InfraID
+	// Copy over the name, cluster ID and infra ID from previous provision so that a failed install can be removed.
+	if lastFailedProvision != nil {
+		provision.Spec.PrevProvisionName = &lastFailedProvision.Name
+		provision.Spec.PrevClusterID = lastFailedProvision.Spec.ClusterID
+		provision.Spec.PrevInfraID = lastFailedProvision.Spec.InfraID
```

Comment:
I have no clue why we were fetching these details from cluster deployment before. Nothing went wrong during my testing after this change.

Reply:
The previous code implies that we've been copying the infra ID to the ClusterDeployment for each provision. This will affect the followon work we've been discussing to add webhooks to make that a write-once field. (No action, just an observation.)